_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -1234,47 +1236,60 @@ mfxStatus VAAPIVideoCORE::IsGuidSupported(const GUID /*guid*/,
+    MFX_CHECK(va_max_num_entrypoints & va_max_num_profiles, MFX_ERR_UNSUPPORTED);
+
+    //driver always support VAProfileNone
+    if (req_profile != VAProfileNone)
     {
-    case MFX_CODEC_VC1:
-        break;
-    case MFX_CODEC_AVC:
-        break;
-    case MFX_CODEC_HEVC:
-        if (m_HWType < MFX_HW_HSW)
-            return MFX_WRN_PARTIAL_ACCELERATION;
-        if (par->mfx.FrameInfo.Width > 8192 || par->mfx.FrameInfo.Height > 8192)
-            return MFX_WRN_PARTIAL_ACCELERATION;
-        break;
-    case MFX_CODEC_MPEG2:
-        if (par->mfx.FrameInfo.Width  > 2048 || par->mfx.FrameInfo.Height > 2048) //MPEG2 decoder doesn't support resolution bigger than 2K
-            return MFX_WRN_PARTIAL_ACCELERATION;
-        break;
-    case MFX_CODEC_JPEG:
-        if (par->mfx.FrameInfo.Width > 8192 || par->mfx.FrameInfo.Height > 8192)
-            return MFX_WRN_PARTIAL_ACCELERATION;
-        break;
-    case MFX_CODEC_VP8:
-        if (m_HWType < MFX_HW_BDW)
+        vector <VAProfile> va_profiles (va_max_num_profiles, VAProfileNone);
+
+        //ask from driver about profile support
+        VAStatus va_sts = vaQueryConfigProfiles(m_Display,
+                            va_profiles.data(), &va_max_num_profiles);
+        MFX_CHECK(va_to_umc_res(va_sts) == UMC_OK, MFX_ERR_UNSUPPORTED);
+
+        //check profile support
+        auto it_profile = find(va_profiles.begin(), va_profiles.end(), req_profile);
+        if (it_profile == va_profiles.end())
             return MFX_ERR_UNSUPPORTED;
-    case MFX_CODEC_VP9:
-        break;
-    default:
-        return MFX_ERR_UNSUPPORTED;
     }
 
-    if (MFX_HW_SNB == m_HWType)
-    {
-        if (par->mfx.FrameInfo.Width > 1920 || par->mfx.FrameInfo.Height > 1200)
-            return MFX_WRN_PARTIAL_ACCELERATION;
-    }
-    else
-    {
-        if (MFX_CODEC_JPEG != par->mfx.CodecId &&
-            MFX_CODEC_HEVC != par->mfx.CodecId &&
-        (par->mfx.FrameInfo.Width > 4096 || par->mfx.FrameInfo.Height > 4096))
-            return MFX_WRN_PARTIAL_ACCELERATION;
-    }
+    vector <VAEntrypoint> va_entrypoints (va_max_num_entrypoints, static_cast<VAEntrypoint> (0));
+
+    //ask from driver about entrypoint support
+    VAStatus va_sts = vaQueryConfigEntrypoints(m_Display, req_profile,
+                    va_entrypoints.data(), &va_max_num_entrypoints);
+    MFX_CHECK(va_to_umc_res(va_sts) == UMC_OK, MFX_ERR_UNSUPPORTED);
+
+    //check entrypoint support
+    auto it_entrypoint = find(va_entrypoints.begin(), va_entrypoints.end(), req_entrypoint);
+    if (it_entrypoint == va_entrypoints.end())
+        return MFX_ERR_UNSUPPORTED;
+
+    VAConfigAttrib attr[2] = {{VAConfigAttribMaxPictureWidth, 0},
+                             {VAConfigAttribMaxPictureHeight, 0}};
+
+    //ask from driver about support
+    va_sts = vaGetConfigAttributes(m_Display, mapper.profile,
+                                   mapper.entrypoint,
+                                   attr, 2);
```

Comment:
actually, it is a bad practice to use hardcoded constant. Just use std::size(attr) instead, it will work for arrays, vectors, etc